kv/dist_sender.go

```diff
@@ -397,6 +397,50 @@ func (ds *DistSender) Send(call *client.Call) {
 			} else if isMulti {
 				// If this request spans ranges, collect the replies.
 				responses = append(responses, reply)
+
```

Comment:
This method has gotten rather large. I'd love to find a way to refactor it into more manageable pieces. Could you add a general TODO here?